### DISCUSSION TITLE
Should LinuxSerialDriver Use `guarded_input`

### DISCUSSION THREAD

**Discussion by LeStarch** [nasa/fprime#728] [Q&A] [answered]
Should the LinuxSerialDriver user `guarded_input` in place of the `sync_input` on the "send" port.  As it is now, two sends could collide and it is unclear from the code if `::write` itself is thread safe.

**Comment by LeStarch** [comment, marked as answer]
@timcanham can you weigh-in here?

**Reply by timcanham** [reply]
That seems reasonable if multiple components are using the driver. I think the original mental model is that it would only have one user, so I lock was unnecessary.

**Reply by LeStarch** [reply]
Awesome!

**Reply by LeStarch** [reply]
We'll take an action to update it as it is minor overhead to provide safety.

**Comment by LeStarch** [comment]
Note: we should also make LinuxSerialDriver as a ByteStreamDriverModel.

**Comment by sheershak-agarwal** [comment]
Hello, I am using F-Prime 1.5.3 and using uart in two different components. I also created two different modules for the uartDrv in topology.cpp. Should I be worried about this? Both the components will be using uart frequently. If this is an issue, then is it fixed in the new release?

**Reply by LeStarch** [reply]
No, this is only an issue if multiple components use the same UART instantiation in topology.  It sounds like you have made two separate instantiations in topology and thus this issue shouldn't be a problem.
